src/deepqmc/wf/paulinet/paulinet.py

```diff
@@ -379,9 +386,17 @@ def forward(self, rs):  # noqa: C901
         if fs is not None and self.backflow_type == 'det':
             n_conf = len(self.confs)
             fs = fs.unflatten(1, (fs.shape[1] // n_conf, n_conf))
-            det_up = self._backflow_op(det_up, fs[..., : self.n_up, : self.n_up])
-            det_down = self._backflow_op(det_down, fs[..., self.n_up :, : self.n_down])
-            # with open-shell systems, part of the backflow output is not used
+            if self.full_determinant:
+                det_full = fs.new_zeros((*det_up.shape[:3], n_elec, n_elec))
+                det_full[..., : self.n_up, : self.n_up] = det_up
+                det_full[..., self.n_up :, self.n_up :] = det_down
+                det_up = det_full = self._backflow_op(det_full, fs)
+                det_down = fs.new_empty((*det_down.shape[:3], 0, 0))
+            else:  # part of the backflow output is not used here
```

Comment:
Doesn't that mean the "normal" PauliNet will always discard part of the backflow output? Maybe that doesn't matter computationally, but it feels like adding complexity even though the full determinant doesn't seem to become relevant in practice.

Reply:
In 1fb1fd2684631a79613b96fb46833cbe34e4e432 the determinant backflow is refactored so that this doesn't happe